### PR TITLE
New connection banner for testing

### PR DESCRIPTION
**Pull request by richardmuscat**
Following brief conversation with @dereksmart, I put together this PR as a starting point for testing a new iteration of the connection banner.

The new version should ideally be split tested against the current one like @jeffgolenski did with the first iteration.

What I'm aiming for here is a simplification and hopefully more compelling explanation of Jetpack, building upon some of @jessefriedman's work with JPO.

What I've done in the PR is to create a new function called `render_banner_b()` within `class.jetpack-connection-banner.php` that, once the split testing logic is set up, can be called instead of the default `render_banner()` 50% of the time.

Testing instructions from @ebinnion:

-Checkout PR
- Deactivate Jetpack, and you should see banner on plugins page
- You should be able to go to `$site.com/wp-admin/options.php` and see the value for `jetpack_connection_banner_ab`. When the value is `1` you should see the previous banner. Then the value is `2`, you should see @richardmuscat's changes
- You can update the value to change which test you get, or you can delete the option and reload the page to test the split testing

### CODE REVIEW

**Comment by richardmuscat** [comment]
This is how the new version should look:

connect-1
connect-2
connect-3
connect-4

**Comment by richardmuscat** [comment]
The title "Welcome to WordPress" on the first slide is *intentional*, aimed at positioning Jetpack (particularly for fresh installs) as an indispensable part of WordPress.

**Comment by richardmuscat** [comment]
@ebinnion -- also pinging you because I believe you had done the original AB test I mention here in case you can lend a hand if its needed.

**Comment by jeffgolenski** [comment]
@richardmuscat 🚢 it

File: class.jetpack-connection-banner.php [hunk 1]
```diff
@@ -470,6 +470,251 @@ class="dops-button is-primary">
...
+							esc_html_e(
+								'Connect to WordPress.com (free) to get started.',
+								'jetpack'
+							);
+							?>
+						</p>
+
+						<p class="jp-banner__button-container">
+							<a
+								href="<?php echo esc_url( Jetpack::init()->build_connect_url( true, false, 'banner-44-slide-1' ) ); ?>"
```
Comment (ebinnion) [inline class.jetpack-connection-banner.php]: The 3rd argument here is for the `from` parameter, so we'll want to update this to something like `banner-52-slide-x` so we can compare. Same for the other slides.

File: class.jetpack-connection-banner.php [hunk 2]
```diff
@@ -470,6 +470,251 @@ class="dops-button is-primary">
...
+
+						<div class="jp-wpcom-connect__content-icon">
+							<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" viewBox="2 3 100 100" version="1.1"><defs><polygon points="36 22.5 36 0 0 0 0 22.5 0 45 36 45"/></defs><g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" transform="translate(2.000000, 3.000000)"><circle fill="#3D596D" cx="50" cy="50" r="50"/><g transform="translate(32.000000, 25.000000)"><mask fill="white"/><path d="M20.3 30.9L20.3 36 15.8 36 15.8 30.9C14.4 30.1 13.5 28.7 13.5 27 13.5 24.5 15.5 22.5 18 22.5 20.5 22.5 22.5 24.5 22.5 27 22.5 28.7 21.6 30.1 20.3 30.9L20.3 30.9ZM11.3 11.2C11.3 7.5 14.3 4.5 18 4.5 21.7 4.5 24.8 7.5 24.8 11.2L24.8 13.5 11.3 13.5 11.3 11.2ZM31.5 13.5L29.3 13.5 29.3 11.2C29.3 5 24.2 0 18 0 11.8 0 6.8 5 6.8 11.2L6.8 13.5 4.5 13.5C2 13.5 0 15.5 0 18L0 40.5C0 43 2 45 4.5 45L31.5 45C34 45 36 43 36 40.5L36 18C36 15.5 34 13.5 31.5 13.5L31.5 13.5Z" fill="#FFFFFF" mask="url(#mask-2)"/></g></g></svg>
+						</div>
+
+						<p>
+							<?php
+							esc_html_e(  'Jetpack checks your site every few minutes and if its offline we will notify you instantly.',
```
Comment (ebinnion) [inline class.jetpack-connection-banner.php]: Formatting is off here. We should drop the first arg down a line probably. Same for below.

File: class.jetpack-connection-banner.php [hunk 3]
```diff
@@ -470,6 +470,251 @@ class="dops-button is-primary">
 	}
 
 	/**
+	 * Renders a modified connection banner.
+	 *
+	 * @since 4.4.0
```
Comment (ebinnion) [inline class.jetpack-connection-banner.php]: This should be updated to `5.3.0`

**Comment by ebinnion** [comment]
I just pushed copy updates:

screen shot 2017-08-10 at 12 24 29 pm

screen shot 2017-08-10 at 12 24 38 pm

screen shot 2017-08-10 at 12 24 44 pm

screen shot 2017-08-10 at 12 24 50 pm

**Comment by ebinnion** [comment]
I briefly chatted with Jeff today, and we're going to want to come up with a better method for running the split test. I'll think about that a bit.

**Comment by keoshi** [comment]
Looking good but noticed a couple of things we could polish up:

- Replace the images icon on _Create Your Site_ tab with the site icon.
- Right align all illustrations on the main block for consistency.

**Comment by jeffgolenski** [comment]
@ebinnion I'm going to do a little design tinkering on this today. Just a heads up!

**Comment by jeffgolenski** [comment]
I just did some work updating the illustrations on the side of the content to match our wpcom illustrations that we're using everywhere. 

One thing I noticed while testing is that the blue CTA for the first slide says "Sign Up" while the rest of them say "Connect to WordPress.com" — the inconsistency may lead to some confusion amongst new users. Is there a particular reason this was done this way? 

Also, in the navigation of the slides, the first slide says "Welcome to Jetpack" but the header of that slide says "Welcome to WordPress" — I think that's also a bit confusing. 

**Visuals with new illustrations:**
screen shot 2017-08-16 at 1 03 22 pm
screen shot 2017-08-16 at 1 03 28 pm
screen shot 2017-08-16 at 1 03 33 pm
screen shot 2017-08-16 at 1 03 38 pm


**Example of Mobile:**
screen shot 2017-08-16 at 1 03 48 pm


cc @ebinnion @nicoleckohler @richardmuscat

**Comment by nicoleckohler** [comment]
@jeffgolenski Nice catch -- I think:

- All the CTA buttons should say "Connect to WordPress.com"
- The first screen header should read "Welcome to Jetpack"

**Comment by jeffgolenski** [comment]
@nicoleckohler I made your suggested adjustments. However, I can't help but think Richard planned it that way for a reason. tehehehe.

**Comment by eliorivero** [comment]
Is it possible to vertically center the icons? They seem to be aligned towards left and the lock icon looks misaligned.

**Comment by nicoleckohler** [comment]
Looks good to me.

**Comment by jeffgolenski** [comment]
@eliorivero great find, sir!

**Comment by jeffgolenski** [comment]
@nicoleckohler thank you so much!

**Comment by dereksmart** [comment]
This will need a rebase now that #7603 has been merged

**Comment by ebinnion** [comment]
Here are some screenshots after fixing the lock icon alignment:

Old banner:

screen-shot-2017-08-18-at-3 16 56-pm

New banner:

screen-shot-2017-08-18-at-3 16 35-pm

**Comment by eliorivero** [comment]
Alignment looks fantastic, Eric 👍

**Comment by jeffgolenski** [comment]
@ebinnion Thanks for hopping in and taking care of that!

File: class.jetpack-connection-banner.php [hunk 4]
```diff
@@ -424,13 +465,250 @@ class="dops-button is-primary">
...
+							esc_html_e(
+								'Connect to WordPress.com (free) to get started.',
+								'jetpack'
+							);
+							?>
+						</p>
+
+						<p class="jp-banner__button-container">
+							<a
+								href="<?php echo esc_url( Jetpack::init()->build_connect_url( true, false, 'banner-53-slide-1' ) ); ?>"
```
Comment (dereksmart) [inline class.jetpack-connection-banner.php]: In `render_banner()`, we're appending `$current_screen->base` to the tracks string so we can determine which page it was clicked on (It's now displaying on both the dashboard and plugins page now).  Let's add that in this banner as well.

**Comment by richardmuscat** [comment]
❤️

**Comment by ebinnion** [comment]
@dereksmart - I've addressed your feedback. There is now a `build_connect_url_for_slide()` method that makes generating the URL with `from` a bit more DRY.

**Comment by dereksmart** [comment]
@ebinnion good call, it looks great.

**Comment by richardmuscat** [comment]
@ebinnion -- how will we measure the results?

**Comment by ebinnion** [comment]
Since this is pre-connection, the only way we can measure results is via the `from` prop on `jetpack_connection_view` tracks or MC stats. The previous version will have a from prop like `banner-44*` and your banner will have a from prop like `banner-53*`.

**Pull request merged**